File: 1512.Number of Good Pairs.cc
// https://leetcode.com/problems/number-of-good-pairs/

static bool x=[](){
    std::ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int counter = 0;
        for (auto it = nums.begin(); it != nums.end(); ++it)
        {
            counter += count(it + 1, nums.end(), *it);
        }
        return move(counter);
    }
};

File: leetcode/1501-1800/1501-1550/1512.Number of Good Pairs.rs
// https://leetcode.com/problems/number-of-good-pairs

impl Solution {
    pub fn num_identical_pairs(nums: Vec<i32>) -> i32 {
        let mut counter = [0;101];
        for n in nums {
            counter[n as usize] += 1
        }

        let mut sum = 0;
        for c in counter {
            sum += c * (c - 1) / 2;
        }

        sum
    }
}